internal/controller/capacity.go

```diff
@@ -109,8 +110,8 @@ func sortGPUs(updatedInstaSliceObject *inferencev1alpha1.Instaslice) []string {
 	return gpuUUIDs
 }
 
-// accounting logic that finds the correct GPU and index where a slice could be placed.
-func (*InstasliceReconciler) getStartIndexFromPreparedState(instaslice *inferencev1alpha1.Instaslice, gpuUUID string, profileName string) int32 {
+// gpuAllocatedSlices stores already allocated slices in indexes
+func gpuAllocatedSlices(instaslice *inferencev1alpha1.Instaslice, gpuUUID string) []int32 {
```

Comment:
Incorrect accounting: I recommend podallocationresults not be used for calculating allocated slices. This is too late as we already capture intent in podallocationrequests, so podallocationrequests should be used. For accounting logic, I think it is better to do this inside `getStartIndexFromPreparedState` we already have InstaSliceobject passed in it, this could be a function call if the method gets too long

Reply:
You are correct; it does not matter much for `getStartIndexFromPreparedState` since it is only one function and one place that drives allocation. However, now we have a second function that is also trying to determine the allocation state, which is called in multiple places. I think a scenario where podallocationrequests exist, but there are no podallocationresults will lead us to do double counting. Wdyt?

Reply:
If podallocationresults does not exist yet, it does not make sense to update the metrics, as the metrics are intended for slices that are already allocated. That was one of the confusions I had when the API changed and the split happened.